### PR TITLE
[WebNN EP] Allow 0D input/output for Reshape and Expand

### LINKED ISSUE #22215: [WebNN EP] Support reshape and expand with 0d input and 0d output
### Describe the issue

A scalar input to a scalar output should be valid nop for reshape and expand.

### To reproduce

Run onnx model with reshape or expand, the input and output are both 0d scalar on WebNN EP.

### Urgency

_No response_

### ONNX Runtime Installation

Built from Source

### ONNX Runtime Version or Commit ID

a47254eaef87a4572078d3165f4b9d1c4b4c52b1

### Execution Provider

Other / Unknown

### PR DESCRIPTION
**Pull request by Honry**
- Allows Expand input be a scalar
- Allows Reshape input be a scalar
- Allows Reshape to a scalar

Fixed #22215

### CODE REVIEW

**Comment by Honry** [comment]
@fdwr, @mingmingtasd, PTAL, thanks!

**Review by guschmue — DISMISSED** [review]

**Comment by Honry** [comment]
@guschmue, is this ready for merge?

File: onnxruntime/core/providers/webnn/builders/impl/reshape_op_builder.cc [hunk 1]
```diff
@@ -92,24 +101,11 @@ bool ReshapeOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& initializer
...
-    return false;
-
-  if (input_shape.empty()) {
-    LOGS(logger, VERBOSE) << "Reshape does not support empty input shape";
-    return false;
-  }
 
   // WebNN reshape does not support 0 as dimension.
   NodeAttrHelper helper(node);
   const bool allow_zero = helper.Get("allowzero ", 0) == 1;
```
Comment (fdwr) [inline onnxruntime/core/providers/webnn/builders/impl/reshape_op_builder.cc:107]:
```suggestion
  const bool allow_zero = helper.Get("allowzero", 0) == 1;
```

**Review by fdwr — DISMISSED** [review]
👍

**Review by fdwr — APPROVED** [review]
👍

**Review by fdwr — APPROVED** [review]
🐪👍

**Review by fdwr — APPROVED** [review]

**Comment by fdwr** [comment]
@guschmue Can you reapprove? GH doesn't like that I made a typo fix, and so my approval doesn't matter.

**Review by guschmue — APPROVED** [review]

**Pull request merged**